File: python/websocket.py
'''
requirements.txt
'''
import asyncio
import json
import uuid
import websockets


async def receives(uri, payload):
    """Starts infinite websocket connection"""
    async with websockets.connect(uri) as websocket:
        await websocket.send(payload)
        while True:
            response = await websocket.recv()
            print(response)


if __name__ == "__main__":
    """
    you may construct payload whatever you want to subscribe.
    PAYLOAD is constant payload now.
    """
    PAYLOAD = json.dumps(
        [
            {'ticket': str(uuid.uuid4())},
            {'type': 'ticker', 'codes': ['KRW-BTC', 'BTC-ADA']}
        ])
    asyncio.get_event_loop().run_until_complete(
        receives('wss://api.upbit.com/websocket/v1', PAYLOAD))

File: node/lib/websocket.js
var exports = module.exports = {};
var WebSocket = require('ws');
var ws = new WebSocket('wss://api.upbit.com/websocket/v1');
ws.binaryType = 'arraybuffer'

exports.connect = function (type, targets) {
  ws.on('open', function open() {
    const uuid = guid();
    var msg = [{ticket: uuid}, {type: type, codes: targets}];
    ws.send(JSON.stringify(msg));
  });

  ws.on('message', function incoming(data) {
    console.log(String.fromCharCode.apply(null, new Uint8Array(data)));
  });
}

function guid() {
  function s4() {
    return Math.floor((1 + Math.random()) * 0x10000)
      .toString(16)
      .substring(1);
  }
  return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() + s4() + s4();
}
